### ISSUE TITLE
Add train/test split script for phylogenetic trees

### ISSUE THREAD

**Issue by trvrb** [blab/trajectories#2]
## Goal

Create a new script `scripts/train_test_split.py` that marks entire clades as "test" data in an Auspice JSON file, enabling proper train/test splits where test data is truly out-of-sample (not sharing evolutionary paths with training data).

## Motivation

Individual trajectories largely overlap (sharing evolutionary paths), so splitting individual trajectories wouldn't create truly out-of-sample test data. By marking entire clades as "test", we ensure test data represents complete evolutionary lineages that are separate from training data.

## Algorithm

1. Load Auspice JSON and build helper data structures (node map, parent map)
2. Randomly select a seed tip from available (unmarked) tips
3. Walk back from the seed tip toward the root, counting mutations on each branch
4. When accumulated mutations >= `mutations_back`, use that ancestor as the clade root
5. **Check clade size**: If ancestor's clade contains > `max_clade_proportion` of total tips, skip this seed tip and try another (prevents selecting huge clades like "all of Omicron")
6. Mark all descendants of the ancestor as "test"
7. Repeat steps 2-6 until target proportion of tips are marked as test
8. Add `train_test` coloring to `meta.colorings` with "train" (blue) and "test" (red)
9. Add `train_test: {value: "train"|"test"}` to each node's `node_attrs`
10. Write modified Auspice JSON

## CLI Interface

```bash
python scripts/train_test_split.py \
    --json INPUT.json \
    --output OUTPUT.json \
    --test-proportion 0.1 \
    --mutations-back 5 \
    --max-clade-proportion 0.1 \
    --gene nuc \
    --seed 42
```

| Argument | Default | Description |
|----------|---------|-------------|
| `--json` | required | Input Auspice JSON file |
| `--output` | required | Output Auspice JSON file |
| `--test-proportion` | 0.1 | Target proportion of tips as test (0.0-1.0) |
| `--mutations-back` | 5 | Mutations to walk back from seed tip |
| `--max-clade-proportion` | 0.1 | Max size of any single test clade as proportion of total tips |
| `--gene` | "nuc" | Gene key for counting mutations |
| `--seed` | None | Random seed for reproducibility |

## Output Format

The modified Auspice JSON will have:

1. New coloring in `meta.colorings`:
```json
{
  "key": "train_test",
  "title": "Train/Test Split",
  "type": "categorical",
  "scale": [["train", "#4C78A8"], ["test", "#E45756"]]
}
```

2. Each node's `node_attrs` updated with `train_test: {value: "train"}` or `{value: "test"}`

## Edge Cases

- **Oversized clades**: Skip seed tips that lead to clades exceeding `max_clade_proportion`
- **Walking past root**: Use root as ancestor (likely rejected by max clade size check)
- **Overlapping clades**: Set deduplication handles multiple selections of same nodes
- **Small trees**: Ensure at least 1 test tip with `max(1, int(total * proportion))`
- **Cannot reach target**: Stop and report achieved proportion if all candidates lead to oversized clades

## Verification

1. Run on test dataset and check ~10% of tips marked as test
2. View in Auspice - color by "Train/Test Split" should show contiguous red (test) clades
3. Verify multiple distinct test clades exist (not just one giant clade)
